pkg/controller/restore_controller.go

```diff
@@ -462,43 +461,97 @@ func mostRecentCompletedBackup(backups []*api.Backup) *api.Backup {
 	return nil
 }
 
-func (c *restoreController) fetchBackup(objectStore cloudprovider.ObjectStore, name string) (*api.Backup, error) {
-	backup, err := c.backupLister.Backups(c.namespace).Get(name)
-	if err == nil {
-		return backup, nil
+// fetchBackupInfo checks the object storage for a backup that matches the given name. If it doesn't
+// find it, it tries to retrieve it from one of the backup storage locations.
+func (c *restoreController) fetchBackupInfo(backupName string, pluginManager plugin.Manager) (backupInfo, error) {
+	var info backupInfo
+	var err error
+	info.backup, err = c.backupLister.Backups(c.namespace).Get(backupName)
+	if err != nil {
```

Comment:
you removed the code that checks to see if the returned error is a 404 or not.  I'd say that if we get a non-nil, non-404 error, we should return it, rather than processing to the "fallback" code

How about organizing the logic in this order:

1. non-nil, non-404 error (i.e. `err != nil && !apierrors.IsNotFound(err)`) -- return the error
2. no error - get the BackupStorageLocation from the lister, get an object store for the location, return a `*backupInfo`
3. if we get here it means we got a 404 error - so call and return the results of `fetchFallBack`